source/cipher_openssl.c

```diff
@@ -168,6 +179,7 @@ int aws_cryptosdk_md_finish(struct aws_cryptosdk_md_context *md_context, void *o
 }
 
 void aws_cryptosdk_md_abort(struct aws_cryptosdk_md_context *md_context) {
+    AWS_PRECONDITION(!md_context || aws_cryptosdk_md_context_is_valid(md_context));
```

Comment:
This looks odd to me, if only `md_context != NULL` is an acceptable argument, why bother to check its validity?

Reply:
It's saying that either NULL, or a valid object, are OK.